src/components/form-elements/checkbox/Checkbox.jsx

```diff
@@ -0,0 +1,245 @@
+  /**
+   * To be displayed below checkbox and its label. The description is not clickable. You can either pass text or your own component with custom styling.
+   * @example <Checkbox description="More detailed description about this element. You can use here even some formatting and links." />
+   */
+  description?: React.Node | string,
+  /**
+   * Sets whether the checkbox is disabled.
+   * @example <Checkbox disabled />
+   */
+  disabled?: boolean,
+  /**
+   * To be displayed below checkbox and its label. The error message is not clickable. Note: you have to set `invalid` prop as well, to render the error message.
+   * @example <Checkbox invalid errorMessage="Error message">Label</Checkbox>
+   */
+  errorMessage?: string,
+  /**
+   * ID assigned to the checkbox input. If not provided, random id will be generated.
+   * @example <Checkbox id="my-checkbox-1" />
+   */
+  id?: string,
+  /**
+   * Sets whether the checkbox is displayed as indeterminate. Note: this prop doesn't modify the `checked` property.
+   * @example <Checkbox indeterminate />
+   */
+  indeterminate?: boolean,
+  /**
+   * Sets whether the checkbox marked as invalid.
+   * @example <Checkbox invalid />
+   */
+  invalid?: boolean,
+  /**
+   * The name of the checkbox input.
+   * @example <Checkbox name="name" />
+   */
+  name?: string,
+  /**
+   * Function called with an object containing the react synthetic event, whenever the state of the checkbox changes.
+   */
+  onChange: (SyntheticInputEvent<HTMLInputElement>) => void,
+  /**
+   * Sets whether the checkbox input is marked as required. This doesn't affect checkbox style.
+   * @example <Checkbox required />
+   */
+  required?: boolean,
+  /**
+   * Value of the checkbox input.
+   * @example <Checkbox value="1" />
+   */
+  value?: string,
+  /**
+   * ID of a custom label, that describes the checkbox input.
+   * @example <Checkbox aria-labelledby="my-custom-label" />
+   */
+  'aria-labelledby'?: string,
+  ...
+};
+
+const Checkbox = ({
+  checked,
+  children,
+  className,
+  color = 'dark',
+  defaultChecked = false,
+  description,
+  disabled = false,
+  errorMessage,
+  id,
+  indeterminate = false,
+  invalid = false,
+  required = false,
+  name,
+  onChange,
+  value,
+  'aria-labelledby': ariaLabelledBy,
+  ...props
+}: CheckboxPropsType) => {
+  const {current: checkboxId} = React.useRef(
+    id === undefined || id === '' ? generateRandomString() : id
+  );
+  const isControlled = checked !== undefined;
+
+  const [isChecked, setIsChecked] = React.useState(
+    isControlled ? checked : defaultChecked
+  );
+  const inputRef = React.useRef(null);
+
+  React.useEffect(() => {
+    inputRef.current.indeterminate = indeterminate;
+  }, [inputRef, indeterminate]);
+
+  React.useEffect(() => {
+    if (isControlled) setIsChecked(checked);
+  }, [checked, isControlled]);
+
+  const onInputChange = React.useCallback(
+    e => {
+      if (isControlled) setIsChecked(!isChecked);
```

Comment:
this is quaite dangerous if there are other places which use `isChecked`, better use `setIsChecked(val => !val);`